kythe/typescript/indexer.ts

```diff
@@ -32,6 +32,24 @@ export interface VName {
   language: string;
 }
 
+/**
+ * A indexer plugin adds extra functionality with the same inputs as the base
+ * indexer.
+ */
+export interface Plugin {
+  name: string;
+  index:
+      (/**
+        * The VName for the CompilationUnit, containing compilation-wide info.
+        */
+       compilationUnit: VName,
+       /**
+        * A map of path to path-specific VName.
+        */
+       pathVNames: Map<string, VName>, paths: string[], program: ts.Program,
```

Comment:
Instead of replicating the component pieces of the VName construction, what if instead the Plugin took a function that given a path, returned the VName?

Reply:
Great idea. Thanks for the feedback.